Adds support for Private Messaging for users on a pod

prologic opened the pull request:
##### Summary

This basically implements a form of "Direct Messaging" or "Priaving
Messaging", we'll call it the later.

Right now this only implements support for "on-pod" messaging. Cross-pod
will come later when we work out how to "mutually authenticate" between
pods.

The underlying mechanisms are:

- SMTP for inter-pod delivery and user access (_only the later of which
  is tested_).
- POP3 for message access by authenticated users.
- Mailbox file format for storage

And of course there is a Web Interface that wraps all this in a nice
way. An API will come later when we start to think about adding this to
the Mobile App (Goryon).

##### Component Name

- area/backend
- area/ui

##### Test Plan

Tested locally extensively as it was develoepd.

Mostly independent code so existing features are not affected in any
major way and should continue to work as expected.

##### Additional Information

aaadonai commented on internal/messages_handler.go.orig:
```diff
@@ -0,0 +1,112 @@
+
+		ctx.Messages = Messages{
+			&Message{
+				From: "kate",
```
Is this testing data?

prologic replied:
Oh good catch!

aaadonai commented on internal/messages_handler.go.orig:
```diff
@@ -0,0 +1,112 @@
+		ctx := NewContext(s.config, s.db, r)
+
+		if r.Method == http.MethodPost {
+			// TODO: Implement this :D
```
Is this missing or will come later?

prologic replied:
Nice catch! This got moved to another handler

aaadonai commented on internal/messages_handler.go.orig:
```diff
@@ -0,0 +1,112 @@
+			return
+		}
+
+		ctx.Title = "Private Message from Kate: Hello"
```
this seems like testing data to me? Is it?

prologic replied:
Yeah I'll check later today

aaadonai commented on internal/messages_handler.go.rej:
```diff
@@ -0,0 +1,101 @@
+***************
```
ok, looks like .rej file and .orig file were committed by mistake? Am I right?

prologic replied:
Looks like it!

aaadonai commented on internal/options.go.orig:
```diff
@@ -0,0 +1,439 @@
+package internal
```
another .orig file.

aaadonai commented on internal/options.go.rej:
```diff
@@ -0,0 +1,34 @@
+***************
```
another .rej file

aaadonai reviewed (approved):
I did my best to review this to the best of my abilities! ;-)

prologic commented:
Thanks!

prologic commented:
@aaadonai Those were all accidentally committed files from bad merge conflict resolution and didn't notice them left around. Fixed and added to `.gitignore` 👍
